components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java

```diff
@@ -115,6 +121,34 @@ public MicroIntegratorRegistry() {
         this.govRegistry = getUri(defaultFSRegRoot, "governance");
 
         initSecurityRepo();
+        initRegistryListener(defaultFSRegRoot);
+    }
+
+    private void initRegistryListener(String regRoot) {
+        try {
+            MicroIntegratorRegistryListener watcher = new MicroIntegratorRegistryListener(Paths.get(regRoot));
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            executor.submit(watcher);
+            executor.shutdown();
+        } catch (IOException e) {
+            log.error(e);
```

Comment:
Don't we need to throw an exception and interrupt the server startup?

Reply:
+1. Fixed